src/Config/GlobalQuery.ts

```diff
@@ -0,0 +1,32 @@
+import { Query } from '../Query/Query';
+
+export class GlobalQuery {
+    private static instance: GlobalQuery;
+
+    private static readonly empty = '';
+    private _value = GlobalQuery.empty;
+
+    public static getInstance(): GlobalQuery {
+        if (!GlobalQuery.instance) {
+            GlobalQuery.instance = new GlobalQuery();
+        }
+
+        return GlobalQuery.instance;
+    }
+
+    public set(value: string) {
+        this._value = value;
+    }
+
+    public get() {
+        return this._value;
+    }
+
+    public query(): Query {
+        return new Query({ source: this._value });
+    }
+
+    public isEmpty() {
+        return this._value === GlobalQuery.empty;
```

Comment:
I touch on this below, but it would be good to see a new file `GlobalQuery.test.ts` that tests the logic of this class.

For example, I am not really sure of the value of `GlobalQuery.empty`, but the addition of the `isEmpty()` method that does not trim the query has led to an unwanted change in behaviour....

Having discovered this, it would be good to see tests of even this seemingly-very-simple logic, as it has consequences for the behaviour that the user sees...

Reply:
Somewhere there needs to also be code that tests how the settings are now saved and loaded...